editoast/src/models/train_schedule.rs

```diff
@@ -146,6 +164,7 @@ pub struct TrainScheduleSummary {
     pub mechanical_energy_consumed: MechanicalEnergyConsumedBaseEco,
     pub stops_count: i64,
     pub path_length: f64,
+    pub update_required: Option<Vec<TrainScheduleValidation>>,
```

Comment:
Isn't it weird to have an `Option<Vec>` ? The "no problem" situation could be represented by an empty `Vec` instead of a None. Else you have to guarantee everywhere that when `update_required` is `Some`, the list is not empty.

Reply:
@multun suggested the Option<Vec> and I find it a good solution first.  
However I agree with you, I guess the empty list is a better option, even for the front. 
What do you think @multun  ?